expression-language.md

```diff
@@ -0,0 +1,369 @@
+| `x / y: Integer x Integer -> Integer`   | Returns the rounded division of `x` and `y`. Returns `0` if `y = 0` and raise an error          |
+| `x % y: Integer x Integer -> Integer`   | Returns the remainder of the division of `x` and `y`. Returns `0` if `y = 0` and raise an error |
+| `x + y: Integer x Integer -> Integer`   | Returns the sum of `x` and `y`                                                                  |
+| `x - y: Integer x Integer -> Integer`   | Returns the difference of `x` and `y`                                                           |
+| `x = y: String x String -> Boolean`     | Returns `true` if the values of `x` and `y` are equal                                           |
+
+#### 3.4.3. Like operator
+
+| Definition                                       | Semantics                                           |
+| ------------------------------------------------ | --------------------------------------------------- |
+| `x LIKE pattern: String x String -> Boolean`     | Returns `true` if the value x matches the `pattern` |
+| `x NOT LIKE pattern: String x String -> Boolean` | Same as `NOT (x LIKE PATTERN)`                      |
+
+The pattern of the `LIKE` operator can contain:
+
+- `%` represents zero, one, or multiple characters
+- `_` represents a single character
+
+For example, the pattern `_b*` will accept values `ab`, `abc`, `abcd1` but won't accept values `b` or `acd`.
+
+Both `%` and `_` can be escaped with `\`, in order to be matched literally. For example, the pattern `abc\%` will match
+`abc%` but won't match `abcd`.
+
+#### 3.4.4. Exists operator
+
+| Definition                          | Semantics                                                                   |
+| ----------------------------------- | --------------------------------------------------------------------------- |
+| `EXISTS identifier: Any -> Boolean` | Returns `true` if the attribute `identifier` exists in the input CloudEvent |
+
+Note: `EXISTS` MUST always return `true` for the required context attributes because the input CloudEvent is always
+assumed valid, e.g. `EXISTS id` must always return `true`.
+
+#### 3.4.5. In operator
+
+| Definition                                             | Semantics                                                                |
+| ------------------------------------------------------ | ------------------------------------------------------------------------ |
+| `x IN (y1, y2, ...): String x String^n -> Boolean`     | Returns `true` if `x` is an element included in the set of `yN` elements |
+| `x NOT IN (y1, y2, ...): String x String^n -> Boolean` | Same as `NOT (x IN set)`                                                 |
+
+The set of values `yN` can be of arbitrary length, and the matching is done using the same semantics of the equal `=`
+operator.
+
+### 3.5. Functions
+
+The following tables show the built-in functions that MUST be supported by a CESQL evaluator.
```

Comment:
UDFs will be too heavy?

Reply:
Definitely too heavy and out of the scope. Note: the spec allows vendors to extend the set of built-ins, so I assume an engine can have a way to add custom functions.

But the language per-se doesn't allow function definition, it's too heavy and it might lead us into the turing completeness trap.